packages/flutter/test/material/material_button_test.dart

```diff
@@ -67,4 +274,403 @@ void main() {
+        padding: const EdgeInsets.all(4.0),
+      ),
+    );
+    expect(tester.getSize(find.byType(MaterialButton)), const Size(8.0, 8.0));
+
+    // Size is defined by the padded child.
+    await tester.pumpWidget(
+      buildFrame(
+        minWidth: 0.0,
+        height: 0.0,
+        padding: const EdgeInsets.all(4.0),
+        child: const SizedBox(width: 8.0, height: 8.0),
+      ),
+    );
+    expect(tester.getSize(find.byType(MaterialButton)), const Size(16.0, 16.0));
+
+    // Size is defined by the minWidth, height constraints.
+    await tester.pumpWidget(
+      buildFrame(
+        minWidth: 18.0,
+        height: 18.0,
+        padding: const EdgeInsets.all(4.0),
+        child: const SizedBox(width: 8.0, height: 8.0),
+      ),
+    );
+    expect(tester.getSize(find.byType(MaterialButton)), const Size(18.0, 18.0));
+  });
+
+  testWidgets('MaterialButton size is configurable by ThemeData.materialTapTargetSize', (WidgetTester tester) async {
+    final Key key1 = UniqueKey();
+    await tester.pumpWidget(
+      Theme(
+        data: ThemeData(materialTapTargetSize: MaterialTapTargetSize.padded),
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: Material(
+            child: Center(
+              child: MaterialButton(
+                key: key1,
+                child: const SizedBox(width: 50.0, height: 8.0),
+                onPressed: () { },
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(tester.getSize(find.byKey(key1)), const Size(88.0, 48.0));
+
+    final Key key2 = UniqueKey();
+    await tester.pumpWidget(
+      Theme(
+        data: ThemeData(materialTapTargetSize: MaterialTapTargetSize.shrinkWrap),
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: Material(
+            child: Center(
+              child: MaterialButton(
+                key: key2,
+                child: const SizedBox(width: 50.0, height: 8.0),
+                onPressed: () { },
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(tester.getSize(find.byKey(key2)), const Size(88.0, 36.0));
+  });
+
+  testWidgets('MaterialButton shape overrides ButtonTheme shape', (WidgetTester tester) async {
+    // Regression test for https://github.com/flutter/flutter/issues/29146
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: MaterialButton(
+          onPressed: () { },
+          shape: const StadiumBorder(),
+          child: const Text('button'),
+        ),
+      ),
+    );
+
+    final Finder rawButtonMaterial = find.descendant(
+      of: find.byType(MaterialButton),
+      matching: find.byType(Material),
+    );
+    expect(tester.widget<Material>(rawButtonMaterial).shape, const StadiumBorder());
+  });
+
+  testWidgets('MaterialButton disabled default is correct.', (WidgetTester tester) async {
```

Comment:
Maybe put this test directly underneath the "MaterialButton elevation and colors have proper precedence", since those are somewhat related. 

I thought about maybe making this part of the above test since it fits that category, but it might be good to keep these tests separate since it was added as a regression test.

I would also change the name of the test to be slightly clearer, like: 
```suggestion
  testWidgets('MaterialButton\'s disabledColor takes precedence over its default disabled color.', (WidgetTester tester) async {
```

Reply:
Agreed